internal/expressions/function/functions.go

```diff
@@ -161,7 +171,8 @@ func FreightMetadata(
 	return expr.Function(
 		"freightMetadata",
 		freightMetadata(ctx, c, project),
-		new(func(freightRefName, key string) any),
+		new(func(freightRefName string) map[string]any),
+		new(func(freightRefName, key string) any), // Deprecated
```

Comment:
[Learned the hard way](https://github.com/akuity/kargo/pull/2998) that when expression functions have overloaded signatures, they should be arranged from longest/most complex to shortest/least complex. It may not matter here, but I have dug deep enough to know if it does or doesn't. I would play it safe and swap the order of these.